### ISSUE TITLE
error creating bean vaultTokenRefresher

### ISSUE THREAD

**Issue by nhomble** [pivotal-cf/spring-cloud-services-starters#128]
I am upgrading my spring boot 2.2 application to the new spring cloud dependencies
```
        <spring.boot.version>2.2.2.RELEASE</spring.boot.version>
        <spring.cloud.services.version>3.1.1.RELEASE</spring.cloud.services.version>
```
and after doing this upgrade, it looks like when the vault refresh scheduler was migrated from the older connector libs into a new auto configuration its use of a final inner class is causing issues. Here is the output from spring-core
```
Application run failed org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'vaultTokenRefresher' defined in class path resource [io/pivotal/spring/cloud/config/client/VaultTokenRenewalAutoConfiguration.class]: Initialization of bean failed; nested exception is org.springframework.aop.framework.AopConfigException: Could not generate CGLIB subclass of class io.pivotal.spring.cloud.config.client.VaultTokenRenewalAutoConfiguration$VaultTokenRefresher: Common causes of this problem include using a final class or a non-visible class; nested exception is java.lang.IllegalArgumentException: Cannot subclass final class io.pivotal.spring.cloud.config.client.VaultTokenRenewalAutoConfiguration$VaultTokenRefresher
```

**Comment by dyroberts** [comment]
@nhomble We haven't run into this issue in our testing. Can you tell me what else might be happening in your app that's causing the framework to try to proxy this bean with cglib? It's an easy enough fix to drop the `final` on the class; but I'm having trouble even reproducing the issue myself.

**Comment by nhomble** [comment]
Thanks @dyroberts - I'll see if I dig a little deeper. Otherwise, I'll try to reproduce this in a smaller project.
Was there any particular reason for the final? Or was it to be extra careful?

**Comment by Walliee** [comment]
A sample application recreating this issue can be found here: https://github.com/Walliee/vault-token-refresher-final-class-issue

The issue surfaces with a particular set of dependencies on the classpath. 
In our case we face it due to having the following dependencies in the project:
```xml
<dependencyManagement>
	<dependencies>
		<dependency>
			<groupId>io.pivotal.spring.cloud</groupId>
			<artifactId>spring-cloud-services-dependencies</artifactId>
			<version>3.1.1.RELEASE</version>
			<type>pom</type>
			<scope>import</scope>
		</dependency>
		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-dependencies</artifactId>
			<version>Hoxton.RELEASE</version>
			<type>pom</type>
			<scope>import</scope>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-dependencies</artifactId>
			<version>2.2.5.RELEASE</version>
			<type>pom</type>
			<scope>import</scope>
		</dependency>
	</dependencies>
</dependencyManagement>

<dependencies>
	<dependency>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-web</artifactId>
	</dependency>
	<dependency>
		<groupId>org.springframework.cloud</groupId>
		<artifactId>spring-cloud-starter-sleuth</artifactId>
	</dependency>
	<dependency>
		<groupId>io.pivotal.spring.cloud</groupId>
		<artifactId>spring-cloud-services-starter-config-client</artifactId>
	</dependency>
</dependencies>
```

More specifically, the problem arises when `AnnotationAwareAspectJAutoProxyCreator` is registered as a `BeanPostProcessor`.

**Comment by Walliee** [comment]
@dyroberts where you able to take a look at the sample?

**Comment by dyroberts** [comment]
Yes! I'm going to remove the `final` in the next release.

**Comment by Walliee** [comment]
This can be closed as it was resolved as part of #139
